### DISCUSSION TITLE
Does NNCF quantize bias parameters to a higher precision?

### DISCUSSION THREAD

**Discussion by i3abghany** [openvinotoolkit/nncf#2080] [General] [answered]
Hello,

In other quantization frameworks, e.g. TFLite(https://arxiv.org/pdf/1712.05877.pdf), Glow (https://github.com/pytorch/glow/blob/a07fbed5da37e6f9bf8b459d15b445067efa3173/docs/Quantization.md?plain=1#L184C28-L184C28), and also in PyTorch (https://discuss.pytorch.org/t/is-bias-quantized-while-doing-pytorch-static-quantization/146416/5), biases are typically either left as FLOAT32 or quantized to higher-precision int datatypes e.g. int32. Does NNCF support a similar thing?

Thank you,
Mahmoud

**Comment by alexsu52** [comment]
Hello @i3abghany ,

NNCF leaves biases in FP32.

**Reply by i3abghany** [reply]
Hello @alexsu52 , 

Thanks for the reply! Is there a way to make such high-precision parameters in integer-only data types?

**Reply by alexsu52** [reply, marked as answer]
Hello @i3abghany,

I think it is possible, but I would like to understand the user scenario and the benefits of this option in NNCF. Could you share your use case? Perhaps I can make some suggestion.

**Reply by i3abghany** [reply]
Hey @alexsu52,
Thanks for following up! There is no specific use case for me. I am just surveying how general-purpose quantization frameworks quantize biases.

**Reply by UsingtcNower** [reply]
> Hello @i3abghany,
> 
> I think it is possible, but I would like to understand the user scenario and the benefits of this option in NNCF. Could you share your use case? Perhaps I can make some suggestion.

Hi @alexsu52 , I want to convert openvino int8 model to onnx, and needs bias as int32. It could be a scenario.

**Reply by alexsu52** [reply]
Hello @UsingtcNower, NNCF supports for quantization onnx model. If you need to get onnx int8 model, you can use `nncf.quantize` and feed the onnx fp32 model as input. If you still want to convert openvino int8 model to onnx then you can cast the float bias to int32 and I don't wait significant drop by accuracy because OpenVINO runtime doing almost the same during inference.
